uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js

```diff
@@ -67,11 +57,7 @@ define([
 
         this.sendGET('LeaveRequest', 'getbalancechangebyabsencetype', params, false)
           .then(function (data) {
-            if (checkError(data)) {
-              deferred.reject(data.error_message);
-            } else {
-              deferred.resolve(data.values);
-            }
+            deferred.resolve(data.values);
```

Comment:
This wont work as we are still returning `deferred.promise`, so we need to resolve.